app/models/topic.rb

```diff
@@ -404,7 +404,9 @@ def self.visible_post_types(viewed_by = nil, include_moderator_actions: true)
     types = Post.types
     result = [types[:regular]]
     result += [types[:moderator_action], types[:small_action]] if include_moderator_actions
-    result << types[:whisper] if viewed_by&.staff?
+    if SiteSetting.enable_whispers.present? && viewed_by&.whisperer?
```

Comment:
This is a change in behaviour since I won't be able to see whispers anymore if I turn off whispers.